src/field.rs

```diff
@@ -12,24 +13,58 @@ pub enum LanguageField {
 }
 
 pub trait LurkField: PrimeField + PrimeFieldBits {
-    // These constants are assumed to be based on some global table like
-    // multicodec, ideally extended to include arbitrary precision codecs
-    const FIELD_CODEC: u64;
-    const HASH_CODEC: u64;
-    const LURK_CODEC_PREFIX: u64 = 0x10de;
-    const NUM_BYTES: usize;
+    const FIELD: LanguageField;
 
+    fn to_bytes(self) -> Vec<u8> {
+        let repr = self.to_repr();
+        repr.as_ref().to_vec()
+    }
     fn from_bytes(bs: &[u8]) -> Option<Self> {
         let mut def: Self::Repr = Self::default().to_repr();
         def.as_mut().copy_from_slice(bs);
         Self::from_repr(def).into()
     }
-    // Return a u32 corresponding to the first 4 little-endian bytes of this field element, discarding the remaining bytes.
-    fn to_u32_unchecked(&self) -> u32 {
-        let mut byte_array = [0u8; 4];
-        byte_array.copy_from_slice(&self.to_repr().as_ref()[0..4]);
-        u32::from_le_bytes(byte_array)
+
+    fn hex_digits(self) -> String {
+        let mut s = String::new();
+        let bytes = self.to_bytes();
+        for b in bytes.iter().rev() {
+            s.push_str(&format!("{:02x?}", b));
+        }
+        s
+    }
+
+    fn vec_f_to_bytes(vec_f: Vec<Self>) -> Vec<u8> {
+        let mut vec = vec![];
+        for f in vec_f {
+            for byte in f.to_bytes() {
+                vec.push(byte)
+            }
```

Comment:
This might be okay, if you allocate the whole capacity up-front. It might be worth optimizing using whatever voodoo (`Vec::from_raw_parts`?) is needed to just use the bytes as they are.